elastic/apm-server
# Decoder rumv3 errors

## Motivation/summary
This PR adds the new decoder handling for `RUM v3` for `metricsets` (top-level and nested inside transactions), `spans` (nested inside transactions) and `errors`. The basic handling for `transactions` and `metadata` has already been implemented. 
It also wires the new decoder to the stream processor and removes obsolete processor logic. 

This PR makes a lot of logic related to the previously used decoding mechanism obsolete, but that logic will be removed in a follow-up PR to keep this one focused on RUM.

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes
Send different RUM v3 payloads to the APM Server and check the ingested events. 
Put a focus on 
* all expected fields showing up in the event. 
* all shortened names being replaced by long names
* metadata information from one event should not be overwritten if merged with eg. labels in another event
* all event types being processed as expected
* APM Server responses being as expected

Careful manual testing is very important here, as automated integration tests for RUM v3 are rather limited. 

## Related issues
part of #3551
